src/language/templates/ast/ast.hpp

```diff
@@ -168,6 +168,23 @@ namespace ast {
             return "{{ node.class_name }}";
         }
 
+        {% if node.nmodl_name %}
+        /**
+        * \brief Return NMODL statement of ast node as std::string
+        *
+        * Every node is related to a special statement in the NMODL. This
+        * statement can be returned as a std::string for printing to
+        * text/json form.
+        *
+        * @return name of the statement as a string i.e. "{{ node.nmodl_name }}"
+        *
+        * \sa Ast::get_nmodl_name
+        */
+        std::string get_nmodl_name() {
+            return "{{ node.nmodl_name }}";
+        }
```

Comment:
Ahh I see that you have already implemented it here. Few comments:

* Not every AST node has associated nmodl name. In this case this is going to return empty string (?).
* I would change that behaviour a bit :  add this `get_nmodl_name` method in top level AST class (see ast_common.hpp, similar to `get_node_name`,
* the nodes which has nmodl_name non empty, they only implement this method

This way the code will throw an error if we try to use `get_nmodl_name` for the node where nmodl name property is not exist in yaml specification.